### ISSUE TITLE
Flow >= 0.58 continuously crashes with SIGSEGV, does not start

### ISSUE THREAD

**Issue by mwiencek** [facebook/flow#5472]
I made some changes to my project in https://github.com/metabrainz/musicbrainz-server/pull/574 which cause `flow` to no longer start; the server monitor logs just show `Flow server (pid ...) was killed with sigsegv signal` repeatedly. This happens on OS X and Linux.

v0.57.3 is the last working version.

I haven't been able to isolate the issue other than this: if I remove `./node_modules/immutable/dist/immutable.js.flow` from `[libs]` in my .flowconfig, it no longer crashes (but is missing those types).

How to reproduce:

```
git clone https://github.com/metabrainz/musicbrainz-server.git
cd musicbrainz-server
npm install
npm install -g flow-bin@0.63.0
flow
```

I've attached a crash report from v0.60.1.

[flow_2017-12-05-124315_Michaels-MacBook-Pro-2.txt](https://github.com/facebook/flow/files/1532478/flow_2017-12-05-124315_Michaels-MacBook-Pro-2.txt)

**Comment by mwiencek** [comment]
Still an issue with 0.63.0.

**Comment by mwiencek** [comment]
In order to be more helpful I set up an opam env to build flow and did a git bisect; the first bad commit is https://github.com/facebook/flow/commit/3255d1b558f96bfd2b39dad63d36d7591e0899cf

**Comment by samwgoldman** [comment]
Interesting. Thanks for bisecting. Can you point me to a rev on the linked github project that reproduces?

**Comment by mwiencek** [comment]
Hi Sam, currently it can be reproduced on the master branch, so the current HEAD works for a specific rev to use: https://github.com/metabrainz/musicbrainz-server/commit/3cab3473b33d3d2b92e6c7bfa026b838f3c4c2cc

**Comment by mwiencek** [comment]
This appears to have been fixed in 0.102.0 (0.101.1 and prior releases still being broken). Unclear how from the changelog.
